templates/common/baremetal/files/NetworkManager-clean-initrd-state.yaml

```diff
@@ -0,0 +1,79 @@
+mode: 0755
+path: "/usr/local/bin/nm-clean-initrd-state.sh"
+contents:
+  inline: |
+    #!/bin/bash
+
+    NM_DEVICES_DIR=/run/NetworkManager/devices
+    NM_CONN_DIR=/run/NetworkManager/system-connections
+
+    [ ! -e "$NM_DEVICES_DIR" ] || [ ! -e "$NM_CONN_DIR" ] && exit
+
+    # Some deployments boot with ip=(dhcp|dhcp6) kernel argument. From this, a
+    # NM dracut module generates a default connection profile. If rd.neednet=1
+    # is in the kernel command line or another dracut module needs the network,
+    # NM is also run by dracut, activating the previously generated profile.
+    # This activation generates NM state which makes that profile to be
+    # re-activated by the NM service. And this has the effect that other
+    # profiles deployed by the user for the same set of devices are ignored.
+
+    # The intention here is to remove the state that was generated with the
+    # activation of those profiles during dracut execution. Then when NM service
+    # runs, the profiles generated by dracut, along with other profiles
+    # deployed, are evaluated towards finding the most appropriate profile to
+    # connect a device with.
+    #
+    # There is an RFE filed against NM that once implemented would make this
+    # script unnecessary:
+    # https://bugzilla.redhat.com/show_bug.cgi?id=2089707
+
+    # check that we are doing this for the expected kernel arguments
+    CMD_LINE_REGEX="(^|\s+)ip=(dhcp|dhcp6|dhcp,dhcp6|dhcp6,dhcp)(\s+|$)"
+    grep -Eqs "$CMD_LINE_REGEX" /proc/cmdline || exit
+
+    for device in "${NM_DEVICES_DIR}"/*; do
+      [ -e "$device" ] || continue
```

Comment:
if we are using `bash` we could use `[[`  

http://mywiki.wooledge.org/BashFAQ/031

Reply:
Any specific reason to change it?

Reply:
Probably not, but `[[` seems to be preferred mostly because of no word splitting, no globbing, bash builtin instead of using `/usr/bin/[`, and internal `||` and `&&`.

```sh
[[ ! (-e "$NM_DEVICES_DIR" && -e "$NM_CONN_DIR" )]] && exit

```

Reply:
Ah ok. I usually prefer to stay in POSIX unless there is a specific need.